### DISCUSSION TITLE
Zero Models Problem

### DISCUSSION THREAD

**Discussion by lequochai26** [sandrofigo/VoxReader#16] [Q&A] [answered]
I have a .vox file generated from MagicaVoxel. It works fine on many different apps. But when i use VoxReader library to read this .vox file. It always 0 Models. I tried on many different file but the result all the same, still 0 Models. Can anyone help me ?

This is my .vox file in MagicaVoxel:
![1](https://user-images.githubusercontent.com/99785334/236621643-09ceb0a1-5f97-4a21-a7a0-450ecb11dbb6.png)

This is my source code (I've checked the path and its all correct).
![2](https://user-images.githubusercontent.com/99785334/236621644-82f00dea-bfe1-40f0-9c9b-e2f4728aa199.png)

This is the result when i execute above codes:
![3](https://user-images.githubusercontent.com/99785334/236621646-13425df1-9c04-4dec-a820-f25b488d0c22.png)

**Comment by sandrofigo** [comment]
@lequochai26 Which version of MagicaVoxel are you using to create the .vox file and can you upload the .vox file so I can have a look at it?

**Reply by lequochai26** [reply]
I'm using MagicaVoxel 0.99.6.4
The vox file: https://www.mediafire.com/file/tk2g2pg174ebphv/castle.vox/file

**Reply by sandrofigo** [reply]
Thank you for providing the .vox file. You are right, with this specific .vox file there are some chunks missing in the file so no models are extracted.

I fixed the file locally by opening it with 0.99.6.4 and saving the file again with the "Save Project As" button at the top.
![image](https://github.com/sandrofigo/VoxReader/assets/2115149/6c15f425-5355-4f68-a8dd-1e888c4dfccb)

After that the file reports the correct model count of 1 with VoxReader.

How do you save/export the file in MagicaVoxel?

**Reply by sandrofigo** [reply]
After a little investigation I could reproduce the behaviour by exporting the file with the ".vox" export function instead of saving the "project".
![image](https://github.com/sandrofigo/VoxReader/assets/2115149/ae0152f0-ce3a-457c-b602-af9eaf21a96f)

If the file is exported this way, some chunks are missing that would normally be used to parse the models. I will add the feature to read "exported" .vox files in the next release.

So the quick fix for you should be to open the .vox file with MagicaVoxel and save the project again with the "Save Project As", instead of exporting to ".vox".
Let me know if that worked for you.

**Reply by sandrofigo** [reply, marked as answer]
@lequochai26 This issue should now be fixed with version v4.1.0 https://github.com/sandrofigo/VoxReader/releases/tag/v4.1.0
